src/ServiceControl.Monitoring/Infrastructure/IKnowAboutEndpoints.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace ServiceControl.Monitoring.Infrastructure
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Timings;
+
+    public interface IKnowAboutEndpoints
+    {
+        IDictionary<string, IEnumerable<EndpointInstanceId>> AllEndpointData();
+    }
+
+    public interface IProvideEndpointMonitoringData
```

Comment:
Should we extend this and `IProvideEndpointInstanceMonitoringData` with metric name. In that way we can make `MonitoredEndpoint` and `MonitoredEndpointInstance` unaware of the metrics.